atomicapp/providers/lib/shipy/dpyexec.py

```diff
@@ -0,0 +1,433 @@
+            return True
+
+        except errors.NotFound:
+            logging.info('Could not find container {}'.format(
+                sane_input['container']))
+            return False
+
+    def version(self, client):
+        """
+        shipy equivalent of docker version.
+        If
+        :param client: docker.Client instance
+        :return: 'version' object
+        version.dpy - docker-py version
+        version.capi - Client API version
+        version.sapi - Server API version
+        version.server - Server version, None if incompatible
+        version.compatible - bool - if client and server are compatible or not
+        """
+
+        version = namedtuple('version', 'dpy capi sapi server compatible')
+
+        try:
+            version_info = client.version()
+            version = version(dpy=dpy_version,
+                              capi=constants.DEFAULT_DOCKER_API_VERSION,
+                              sapi=version_info['ApiVersion'],
+                              server=version_info['Version'],
+                              compatible=True)
+
+        except errors.APIError:
+            version = version(dpy=dpy_version,
+                              capi=constants.DEFAULT_DOCKER_API_VERSION,
+                              sapi=client._retrieve_server_version(),
+                              server=None,
+                              compatible=False)
+
+        logging.info('\ndocker-py: {}\n'
+                     'Client API: {}\n'
+                     'Server API: {}\n'
+                     'Compatibility: {}'.format(version.dpy,
+                                                version.capi,
+                                                version.sapi,
+                                                version.compatible))
+        return version
+
+    def shipy(self, args):
+        """
+        The entrypoint for shipy, takes in the arguments from the user,
+        processes them, refines them, and delegates to respective functions.
+        :param args: user arguments
+        :return: whatever the called function returns
+        """
+
+        "Check if input is from a file"
+        if '--file' in args:
+            f_pos = args.index('--file')
+
+            with open(args[f_pos + 1]) as f:
+                command = f.readline().split()
+
+                if command[0] == 'docker':
+                    args = command[1:]
+                else:
+                    raise SyntaxError(
+                        'The input file has a malformed docker command.')
+
+        shipy_parser = parser.define_parsers()
+        sh_args = shipy_parser.parse_args(args)
+        # set logging level
+        if sh_args.isverbose:
+            logging.basicConfig(stream=sys.stdout, level=logging.DEBUG)
+            logging.debug('Setting logging level to logging.DEBUG')
+        else:
+            logging.basicConfig(stream=sys.stdout, level=logging.INFO)
```

Comment:
Should the program calling shipy be able pass it's own custom logger and shipy starts using that, is that fine?